contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,563 @@
+    * max bounty (the time between setPendingMaxBounty and setMaxBounty)
+    * @param _delay The time period for the delay. Must be at least 2 days.
+    */
+    function setMaxBountyDelay(uint256 _delay)
+    external
+    onlyOwner {
+        if (_delay < 2 days) revert DelayTooShort();
+        generalParameters.setMaxBountyDelay = _delay;
+        emit SetMaxBountyDelay(_delay);
+    }
+
+    /**
+    * @notice Create a new vault
+    * @param _asset The vault's native token
+    * @param _committee The address of the vault's committee 
+    * @param _rewardController The reward controller for the vault
+    * @param _maxBounty The maximum percentage of the vault that can be paid
+    * out as a bounty. Must be between 0 and `HUNDRED_PERCENT`
+    * @param _bountySplit The way to split the bounty between the hacker, 
+    * committee and governance.
+    *   Each entry is a number between 0 and `HUNDRED_PERCENT`.
+    *   Total splits should be equal to `HUNDRED_PERCENT`.
+    *   Bounty larger than 0 must be specified for the hacker (direct or 
+    *   vested in vault's native token).
+    * @param _descriptionHash Hash of the vault description.
+    * @param _bountyVestingParams Vesting params for the bounty
+    *        _bountyVestingParams[0] - vesting duration
+    *        _bountyVestingParams[1] - vesting periods
+    * @param _isPaused Whether to initialize the vault with deposits disabled
+    * @return vault The address of the new vault
+    */
+    function createVault(
+        IERC20 _asset,
+        address _committee,
+        IRewardController _rewardController,
+        uint256 _maxBounty,
+        HATVault.BountySplit memory _bountySplit,
+        string memory _descriptionHash,
+        uint256[2] memory _bountyVestingParams,
+        bool _isPaused
+    ) 
+    external 
+    returns(address vault)
+    {
+        vault = Clones.clone(hatVaultImplementation);
+
+        HATVault(vault).initialize(
+            _rewardController,
+            _bountyVestingParams[0],
+            _bountyVestingParams[1],
+            _maxBounty,
+            _bountySplit,
+            _asset,
+            _committee,
+            _isPaused
+        );
+
+        hatVaults.push(vault);
+
+        emit VaultCreated(
+            vault,
+            address(_asset),
+            _committee,
+            _rewardController,
+            _descriptionHash,
+            _maxBounty,
+            _bountySplit,
+            _bountyVestingParams[0],
+            _bountyVestingParams[1]
+        );
+    }
+
+    /**
+    * @notice change the UI visibility of a vault
+    * only calleable by the owner of the contract
+    * @param _vault the vault to update
+    * @param _visible is this vault visible in the UI
+    * This parameter can be used by the UI to include or exclude the vault
+    */
+    function setVaultVisibility(address _vault, bool _visible) external onlyOwner {
```

Comment:
consider storing this explicitly so it can be references by other contracts, #302

Reply:
Done

Reply:
we talked about that on a call. We have no specific plans but now that createVault is trustless it seems there may be some sense in having an on-chain record of our approval of this particular vault

Reply:
ok. when we talked about it i thought we're also going to use it